### ISSUE TITLE
Blosc: Add to Windows CI; MSVC Issue in 2.7.1

### ISSUE THREAD

**Issue by ax3l** [ornladios/ADIOS2#2745]
Hi,

it looks like Blosc is not yet added to CI tests cc @chuckatkins @pnorbert.

I currently see the following build error with ADIOS 2.7.1 + Blosc 1.21.0 on Windows cc @psychocoderHPC:
```
 adios2/operator/compress/CompressBlosc.h(80,35): error C3646: 'DataHeader': unknown override specifier [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\adios2\adios2_core.vcxproj]
 adios2/operator/compress/CompressBlosc.h(80,26): error C2079: 'adios2::core::compress::CompressBlosc::packed' uses undefined class 'adios2::core::compress::__attribute__' [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\adios2\adios2_core.vcxproj]
 adios2/operator/compress/CompressBlosc.h(93,9): error C2275: 'uint32_t': illegal use of this type as an expression [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\adios2\adios2_core.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30037\include\stdint.h(24): message : see declaration of 'uint32_t' [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\adios2\adios2_core.vcxproj]
 adios2/operator/compress/CompressBlosc.h(93,18): error C2146: syntax error: missing ')' before identifier 'format' [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\adios2\adios2_core.vcxproj]
 adios2/operator/compress/CompressBlosc.h(101,5): error C2059: syntax error: 'public' [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\adios2\adios2_core.vcxproj]
 adios2/operator/compress/CompressBlosc.h(106,1): error C2270: 'GetNumChunks': modifiers not allowed on nonmember functions [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\adios2\adios2_core.vcxproj]
 adios2/operator/compress/CompressBlosc.h(108,1): error C2270: 'IsChunked': modifiers not allowed on nonmember functions [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\adios2\adios2_core.vcxproj]
 adios2/operator/compress/CompressBlosc.h(108,1): error C2882: 'format': illegal use of namespace identifier in expression [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\adios2\adios2_core.vcxproj]
 adios2/operator/compress/CompressBlosc.h(113,1): error C2059: syntax error: '<L_GRAMMAR_end_class>' [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\adios2\adios2_core.vcxproj]
 adios2/operator/compress/CompressBlosc.h(115,1): fatal error C1506: unrecoverable block scoping error [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\adios2\adios2_core.vcxproj]
```

This is likely caused by the `__attribute__(( packed ))` and needs to be
- generalized, e.g. adding for MSVC https://stackoverflow.com/a/3312896/2719194
- tested by CI for all platforms

**Comment by ax3l** [comment]
On Linux, with static `libblosc.a` builds, I also see problems during ADIOS2 builds of the form:
```
/usr/local/lib/libblosc.a(blosc.c.o): In function `blosc_init.part.9':
  blosc.c:(.text+0x43e): undefined reference to `pthread_atfork'
  collect2: error: ld returned 1 exit status
```

**Comment by ax3l** [comment]
Fixed in  #2746

**Comment by ax3l** [comment]
But blosc should be added to Windows CI.
